PageBuilder/Breakpoints/view/adminhtml/web/ts/js/content-type/column-group/preview.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+// ADMIN COLUMN-GROUP HACK:
+//
+// This preview component calculates which "break" element in the
+// column-group get a '.break' class. When added to the right element,
+// the '.break' class ensures that the flexbox wraps its items (columns)
+// to the next row. The native flexbox behavior then takes over by
+// proportionally expanding the row's columns to fill the remaining
+// space on the row.
+//
+// This ensures that the custom 'columnsPerRow' property defined
+// in the view.xml file (for each breakpoint) works as it should.
+//
+// Flexbox does not support row breaks. Hence the hack. It works!
+//
+
+import $ from "jquery";
+import Config from "Magento_PageBuilder/js/config";
+import ColumnGroup from "Magento_PageBuilder/js/content-type-collection";
+import ContentTypeConfigInterface from "Magento_PageBuilder/js/content-type-config.types";
+import ColumnGroupPreview from "Magento_PageBuilder/js/content-type/column-group/preview";
+import ObservableUpdater from "Magento_PageBuilder/js/content-type/observable-updater";
+import events from "Magento_PageBuilder/js/events";
+
+/**
+ * @api
+ */
+export default // @ts-ignore
+class Preview extends ColumnGroupPreview {
+    public widthToAdd: number;
+    public columnsPerRow: number;
+    private columns: any;
+
+    /**
+     *
+     * @param {ContentTypeCollection} contentType
+     * @param {ContentTypeConfigInterface} config
+     * @param {ObservableUpdater} observableUpdater
+     */
+    constructor(
+        contentType: ColumnGroup,
+        config: ContentTypeConfigInterface,
+        observableUpdater: ObservableUpdater,
+    ) {
+        super(contentType, config, observableUpdater);
+
+        events.on(`stage:${this.contentType.stageId}:viewportChangeAfter`, (args: {viewport: string}) => {
+            const viewports = Config.getConfig("viewports");
+            const columnsPerRow = parseFloat(viewports[args.viewport].options.columns.default.columnsPerRow);
+
+            // HACK: For flexbox row breaks defined in columnsPerRow.
+
+            // Get all hr elements that were added by the column in initColumn.
+            const hrNodes = $('.pagebuilder-column-group hr');
```

Comment:
Have you tried to control the width of each column? That should make columns to wrap